test/CarouselSpec.tsx

```diff
@@ -0,0 +1,998 @@
+      onKeyDownSpy.should.have.been.calledOnce;
+    });
+
+    ['ArrowUp', 'ArrowRightLeft', 'Onwards'].forEach((key) => {
+      it('should do nothing for non left or right keys', () => {
+        const onSelectSpy = sinon.spy();
+
+        const { getByTestId } = render(
+          <Carousel activeIndex={1} onSelect={onSelectSpy} data-testid="test">
+            {items}
+          </Carousel>,
+        );
+
+        fireEvent.keyDown(getByTestId('test'), {
+          key,
+        });
+        clock.tick(50);
+        sinon.assert.notCalled(onSelectSpy);
+      });
+    });
+  });
+
+  describe('mouse events', () => {
+    let clock: sinon.SinonFakeTimers;
+
+    beforeEach(() => {
+      clock = sinon.useFakeTimers();
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
+
+    it('should handle a defined mouse over event', () => {
+      const onMouseOverSpy = sinon.spy();
+
+      const { getByTestId } = render(
+        <Carousel
+          activeIndex={1}
+          onMouseOver={onMouseOverSpy}
+          data-testid="test"
+        >
+          {items}
+        </Carousel>,
+      );
+
+      fireEvent.mouseOver(getByTestId('test'));
+      clock.tick(1500);
+      onMouseOverSpy.should.have.been.calledOnce;
+    });
+
+    it('should handle a defined mouse out event', () => {
+      const onMouseOutSpy = sinon.spy();
+
+      const { getByTestId } = render(
+        <Carousel activeIndex={1} onMouseOut={onMouseOutSpy} data-testid="test">
+          {items}
+        </Carousel>,
+      );
+
+      fireEvent.mouseOut(getByTestId('test'));
+      clock.tick(50);
+      onMouseOutSpy.should.have.been.calledOnce;
+    });
+  });
+
+  describe('touch events', () => {
+    let clock: sinon.SinonFakeTimers,
+      renderResult: RenderResult,
+      carousel: HTMLElement,
+      onSelectSpy: sinon.SinonSpy,
+      onTouchStartSpy: sinon.SinonSpy,
+      onTouchMoveSpy: sinon.SinonSpy,
+      onTouchEndSpy: sinon.SinonSpy;
+
+    beforeEach(() => {
+      onSelectSpy = sinon.spy();
+      onTouchStartSpy = sinon.spy();
+      onTouchMoveSpy = sinon.spy();
+      onTouchEndSpy = sinon.spy();
+
+      renderResult = render(
+        <Carousel
+          activeIndex={1}
+          interval={null}
+          onSelect={onSelectSpy}
+          onTouchStart={onTouchStartSpy}
+          onTouchMove={onTouchMoveSpy}
+          onTouchEnd={onTouchEndSpy}
+          touch
+          data-testid="carousel-test"
+        >
+          {items}
+        </Carousel>,
+      );
+
+      carousel = renderResult.getByTestId('carousel-test');
+
+      clock = sinon.useFakeTimers();
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
+
+    function generateTouchEvents(params: {
+      target: HTMLElement;
+      startX: number;
+      endX: number;
+    }): void {
+      const { target, startX, endX } = params;
+
+      fireEvent.touchStart(target, {
+        touches: [new Touch({ identifier: 1, target, clientX: startX })],
```

Comment:
In order to have touch event working on Chrome during local build I have added `new Touch(...)` when firing touch events.

Without `new Touch(...)` I get the following error:
```
Failed to construct 'TouchEvent': Failed to read the 'touches' property from 'TouchEventInit': Failed to convert value to 'Touch'.
```

However this seems to be not supported on headless browser. Any idea on how to fix this?